core/corerepo/pinning.go

```diff
@@ -60,18 +60,9 @@ func Pin(n *core.IpfsNode, ctx context.Context, paths []string, recursive bool)
 
 func Unpin(n *core.IpfsNode, ctx context.Context, paths []string, recursive bool) ([]key.Key, error) {
 
-	dagnodes := make([]*merkledag.Node, 0)
-	for _, fpath := range paths {
-		dagnode, err := core.Resolve(ctx, n, path.Path(fpath))
-		if err != nil {
-			return nil, err
-		}
-		dagnodes = append(dagnodes, dagnode)
-	}
-
 	var unpinned []key.Key
-	for _, dagnode := range dagnodes {
-		k, _ := dagnode.Key()
+	for _, path := range paths {
+		k := key.B58KeyDecode(path)
```

Comment:
i think this won't work with paths like `ipfs pin rm /ipfs/<key>/foo/bar/baz` -- which the core.Resolve is there for. what we need is a resolve that _resolves for the last key_ without necessarily fetching the last object.